File: src/modules/clean_system.rs
use std::process::Command;

use crate::utils::status_msg::done;

pub fn clean_system() {
    run_cmd(
        "podman",
        &["image", "prune", "-af"],
        "Failed to prune podman images",
    );
    run_cmd(
        "podman",
        &["volume", "prune", "-f"],
        "Failed to prune podman volumes",
    );
    run_cmd(
        "flatpak",
        &["uninstall", "--unused"],
        "Failed to uninstall unused flatpaks",
    );
    run_cmd(
        "rpm-ostree",
        &["cleanup", "-bm"],
        "Failed to cleanup rpm-ostree",
    );
}

fn run_cmd(cmd: &str, args: &[&str], msg: &str) {
    Command::new(&cmd).args(&*args).spawn().expect(&msg);
    done(format!("{}", &cmd).as_str());
}

File: cmd/cleanSystem.go
package cmd

import (
	"os/exec"

	"github.com/darkmatterorg/orbit/utils"
	"github.com/spf13/cobra"
)

var cleanSystemCmd = &cobra.Command{
	Use:   "clean-system",
	Short: "Clean up old up unused podman images, volumes, flatpak packages and rpm-ostree content",
	Run: func(cmd *cobra.Command, args []string) {

		utils.Info("Cleaning old podman images")
		if err := exec.Command("podman", "image", "prune", "-af").Run(); err != nil {
			utils.Error("Failed to prune podman images")
		} else {
			utils.Done("Pruned podman images")
		}

		utils.Info("Cleaning old podman volumes")
		if err := exec.Command("podman", "volume", "prune", "-f").Run(); err != nil {
			utils.Error("Failed to prune podman volumes")
		} else {
			utils.Done("Pruned podman volumes")
		}

		utils.Info("Cleaning unused flatpak packages")
		if err := exec.Command("flatpak", "uninstall", "--unused").Run(); err != nil {
			utils.Error("Failed to uninstall unused flatpaks")
		} else {
			utils.Done("Uninstalled unused flatpaks")
		}

		utils.Info("Cleaning up rpm-ostree")
		if err := exec.Command("rpm-ostree", "cleanup", "-bm").Run(); err != nil {
			utils.Error("Failed to cleanup rpm-ostree")
		} else {
			utils.Done("Cleaned up rpm-ostree")
		}
	},
}

func init() {
	RootCmd.AddCommand(cleanSystemCmd)
}
